spec/controllers/email_controller_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+RSpec.describe EmailsController, type: :controller do
+  before :each do
+    @event = FactoryGirl.create(:event)
+    @user = FactoryGirl.create(:user, role: :organizer)
+    sign_in @user
+  end
+
+  describe "GET #show" do
+    it "renders email view" do
+      get :show, event_id: @event.id
+      expect(subject).to render_template(:email)
+    end
+
+    context "with valid accepted applications" do
+      before :each do
+        @application = FactoryGirl.create(:application_letter_accepted, event: @event, user: FactoryGirl.build(:user))
+        @template = FactoryGirl.create(:email_template_accepted)
+      end
+
+      it "sets @email with the email of the accepted application" do
+        get :show, event_id: @event.id, status: :accepted
+        expect(assigns(:email)).to be_a(Email)
+        expect(assigns(:email).recipients).to eq(@application.user.email)
+      end
+
+      it "sets @template with template for accepted emails" do
+        get :show, event_id: @event.id, status: :accepted
+        expect(assigns(:templates)).to eq([@template])
+      end
+    end
+
+    context "with valid rejected applications" do
+      before :each do
+        @application = FactoryGirl.create(:application_letter_rejected, event: @event, user: FactoryGirl.build(:user))
+        @template = FactoryGirl.create(:email_template_rejected)
+      end
+
+      it "sets @email with the email of the accepted application" do
+        get :show, event_id: @event.id, status: :rejected
+        expect(assigns(:email)).to be_a(Email)
+        expect(assigns(:email).recipients).to eq(@application.user.email)
+      end
+
+      it "sets @template with template for accepted emails" do
+        get :show, event_id: @event.id, status: :rejected
+        expect(assigns(:templates)).to eq([@template])
+      end
+    end
+  end
+
+  describe "POST #submit" do
+    context "with valid email pressing the send button" do
+      before :each do
+        @email = FactoryGirl.build(:email).attributes
```

Comment:
It gives only the attributes of the email object as dictionary.
It is needed because the post request does not take the email object